service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RespondentController.java

```diff
@@ -117,4 +149,162 @@ public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+        newRespondents.add(newElement(eventData.getTransformedRespondent()));
+
+        CaseData dummyCaseData = caseData.builder()
+            .respondents1(newRespondents)
+            .build();
+
+        List<String> errors = respondentValidator.validate(dummyCaseData, dummyCaseData, true);
+        return respond(caseDetails, errors);
+    }
+
+    private UUID getFirstOtherId(CaseData caseData) {
+        Set<UUID> additionalOtherIds = nullSafeList(caseData.getOthers().getAdditionalOthers())
+            .stream().map(Element::getId).collect(Collectors.toSet());
+        UUID firstOtherUUID = caseData.getConfidentialOthers().stream().map(Element::getId)
+            .filter(co -> !additionalOtherIds.contains(co)).findFirst()
+            .orElse(UUID.fromString("00000000-0000-0000-0000-000000000000"));
+        return firstOtherUUID;
+    }
+
+    private Element<Other> getSelectedOther(CaseData caseData) {
+        UUID firstOtherUUID = getFirstOtherId(caseData);
+        return othersService.getSelectedOther(caseData,
+            caseData.getOtherToRespondentEventData().getOthersList(), firstOtherUUID);
+    }
+
+    private List<Element<Other>> buildNewAllOthers(CaseData caseData, Element<Other> selectedOther) {
+        List<Element<Other>> listOfNewOtherElement = new ArrayList<>(caseData.getAllOthers());
+        listOfNewOtherElement.removeIf(ele -> Objects.equals(ele.getValue(), selectedOther.getValue()));
+        List<Element<Other>> ret = new ArrayList<>();
+        final List<Element<Other>> confidentialOthers = caseData.getConfidentialOthers();
+        listOfNewOtherElement.forEach(element -> {
+            if (element.getValue().containsConfidentialDetails()) {
+                Element<Other> otherElementWithConfidentialDetails = element(element.getId(), othersService
+                    .addConfidentialDetails(getConfidentialItemToAdd(confidentialOthers, element), element));
+                // copying the representedBy to confidential others
+                element.getValue().getRepresentedBy()
+                    .forEach(uuid -> otherElementWithConfidentialDetails.getValue()
+                        .addRepresentative(uuid.getId(), uuid.getValue()));
+                ret.add(otherElementWithConfidentialDetails);
+            } else {
+                ret.add(element);
+            }
+        });
+        return ret;
+    }
+
+    private Others prepareNewOthers(CaseDetails caseDetails, List<Element<Other>> newAllOthers) {
+        // Setting "confidentialOthers" to caseDetails
+        confidentialDetailsService.addConfidentialDetailsToCase(caseDetails, newAllOthers, OTHER);
+        // Setting "others" to caseDetails from previous built new allOthers collection
+        List<Element<Other>> preparingNewOthers = confidentialDetailsService
+            .removeConfidentialDetails(newAllOthers);
+        preparingNewOthers.forEach(
```

Comment:
check if you need this if you don't need to do the add/remove confidential others steps

Reply:
Resolved by removing the selected other from the confidentialOthers